### PR TITLE
[Feature] Support use histogram statistics estimate row count

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 25 / 159 (15.72%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Bucket.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java|8|132|06.06%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnStatistic.java|6|8|75.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|11|14|78.57%|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/Bucket.java [hunk 1]
```diff
@@ -3,23 +3,23 @@
...
 public class Bucket {
-    private final String lower;
-    private final String upper;
+    private final double lower;
+    private final double upper;
     private final Long count;
     private final Long upperRepeats;
 
-    public Bucket(String lower, String upper, Long count, Long upperRepeats) {
+    public Bucket(Double lower, Double upper, Long count, Long upperRepeats) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/Bucket.java:11]:
```suggestion
    public Bucket(double lower, double upper, Long count, Long upperRepeats) {
```

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/Bucket.java [hunk 2]
```diff
@@ -3,23 +3,23 @@
...
         this.lower = lower;
         this.upper = upper;
         this.count = count;
         this.upperRepeats = upperRepeats;
     }
 
-    public String getLower() {
+    public Double getLower() {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/Bucket.java:18]:
```suggestion
    public double getLower() {
```

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/Bucket.java [hunk 3]
```diff
@@ -3,23 +3,23 @@
...
         return lower;
     }
 
-    public String getUpper() {
+    public Double getUpper() {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/Bucket.java:22]:
```suggestion
    public double getUpper() {
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 116 / 129 (89.92%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|11|14|78.57%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java|74|80|92.50%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Bucket.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Histogram.java|22|22|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnStatistic.java|8|8|100.00%|

**Review by Youngwb — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**